### ISSUE TITLE
Document or remove the limitation on the number of concurrently running mutators

### ISSUE THREAD

**Issue by mraleph** [dart-lang/sdk#54687]
We should decide what we want to do with the currently existing limitation on the number of concurrently running mutators.

We should either make it more known (e.g. document it) or we should remove it. 

We could even add a runtime warning: when we are trying to enter isolate as a mutator and are waiting for too long due to the limit on the number of mutators - we should consider reporting this error. 

I don't think the limitation itself is all that bad for mobile usages - because it does not make sense to spawn so many concurrently running isolates anyway, but it might be causing real issues to server side / batch uses of Dart.

/cc @mkustermann @aam @rmacnak-google @a-siva

**Comment by mkustermann** [comment]
There's a set of different aspects we can document

* Semantically one can run only into trouble if N>limit isolates try to synchronously communicate with each other (e.g. by stepping out of the pure Dart safety zone with FFI)
* If one stays in the pure isolates world: isolates communicate via message passing and run async code / often yield to event loop (which is needed to e.g. receive messages), then there's no risk of deadlocks etc. One can have thousands of isolates in that case.
* A dart program cannot take advantage of more cores than this limit. Though as a language focused more on the client-side, it's currently also not a goal to scale to 100s of cores.
* If there's oversubscription: more isolates want to run than the limit, they do schedule round-robin and we're not guaranteed to distribute CPU fairly among them. If there's no oversubscription (number if isolates ready to run is less that the limit), then we run all ready isolates in parallel and OS distributes CPU cycles (hopefully fairly) among them.

**Comment by mkustermann** [comment]
@mraleph Yes, the list I wrote above is what we can/should document. We can of course phrase it differently and just point out it's due to a technical limitation.

**Comment by mraleph** [comment]
I would like to repurpose this issue to investigate if we could simply lift this limitation. So I am retitling this accordingly.

**Comment by blakeAspentech** [comment]
@rmacnak-google , I posted about this with a reproduction [over on the flutter repo](https://github.com/flutter/flutter/issues/162533#issuecomment-2666772282). I see that your commit says it resolves this issue, but it was only committed last week. Is there a version of flutter/dart that I can switch to to verify your fix on [my example repo](https://github.com/blakeAspentech/bug/tree/main)?
